packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SVGComponents/VisXZoom/VisXZoom.js

```diff
@@ -229,29 +229,28 @@ function VisXZoom({
         zoomRef.current = _zoom
         return (
           <g ref={wheelEventLayer}>
-            <ZoomingComponent
-              initialTransformMatrix={_zoom.initialTransformMatrix}
-              transformMatrix={_zoom.transformMatrix}
-              transform={_zoom.toString()}
-              move={move}
-              {...props}
+            <ZoomEventLayer
+              focusable
+              height={height}
+              onDoubleClick={onDoubleClick}
+              onKeyDown={onKeyZoom}
+              onPointerEnter={onPointerEnter}
+              onPointerDown={panning ? _zoom.dragStart : DEFAULT_HANDLER}
+              onPointerMove={panning ? _zoom.dragMove : DEFAULT_HANDLER}
+              onPointerUp={panning ? _zoom.dragEnd : DEFAULT_HANDLER}
+              onPointerLeave={onPointerLeave}
+              onWheel={onWheel}
+              panning={panning}
+              tabIndex={0}
+              width={width}
             >
-              <ZoomEventLayer
-                focusable
-                height={height}
-                onDoubleClick={onDoubleClick}
-                onKeyDown={onKeyZoom}
-                onPointerEnter={onPointerEnter}
-                onPointerDown={panning ? _zoom.dragStart : DEFAULT_HANDLER}
-                onPointerMove={panning ? _zoom.dragMove : DEFAULT_HANDLER}
-                onPointerUp={panning ? _zoom.dragEnd : DEFAULT_HANDLER}
-                onPointerLeave={onPointerLeave}
-                onWheel={onWheel}
-                panning={panning}
-                tabIndex={0}
-                width={width}
-              />
-            </ZoomingComponent>
+              {cloneElement(props.children, {
+                initialTransformMatrix: _zoom.initialTransformMatrix,
+                transformMatrix: _zoom.transformMatrix,
+                transform: _zoom.toString(),
+                move
+               })}
```

Comment:
A render prop might be preferable here, but I was wary of reopening #6778.
https://react.dev/reference/react/cloneElement

Reply:
I converted the code from `cloneElement` to a child render prop, and didn't regress #6778, so I'll push that up.